server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPModule.java

```diff
@@ -128,13 +132,41 @@ JwtConfiguration providesJwtConfiguration(JMAPConfiguration jmapConfiguration) {
     }
 
     @Singleton
+    public static class UpdateACLListener implements ConfigurationPerformer {
+        private final MailboxManager mailboxManager;
+        private final MailboxUpdateParentAsLookupListener updateParentAsLookupListener;
+
+        @Inject
+        public UpdateACLListener(MailboxManager mailboxManager, MailboxUpdateParentAsLookupListener updateParentAsLookupListener) {
+            this.mailboxManager = mailboxManager;
+            this.updateParentAsLookupListener = updateParentAsLookupListener;
+        }
+
+        @Override
+        public void initModule() {
+            try {
+                mailboxManager.addGlobalListener(updateParentAsLookupListener, null);
```

Comment:
null?

That's not acceptable. Please create a system session here!

Also please consider registring just on the delegatingListener ;-) We should start using the event system separatly from mbx manager.

Reply:
I have problem when separate event out of StoreMailboxManager, I don't know how to refactor MailboxManagerStressTest when it needs to addListener